File: Paradigmas de linguagem de programação/Lista de Exercicio - PLP/Questão 14/Python/main.py
class Configuracao:
    _instance = None

    def new(cls):
        if cls._instance is None:
            cls._instance = super(Configuracao, cls).new(cls)
        return cls._instance

config1 = Configuracao()
config2 = Configuracao()

print(config1 is config2)

File: Paradigmas de linguagem de programação/Lista de Exercicio - PLP/Questão 14/Go/main.go
package main
import (
    "fmt"
    "sync"
)

type Configuracao struct{}

var instance Configuracao
var once sync.Once

func GetInstance()Configuracao {
    once.Do(func() {
        instance = Configuracao{}
    })
    return instance
}

func main() {
    config1 := GetInstance()
    config2 := GetInstance()

    fmt.Println(config1 == config2)  // true
}